src/system.cr

```diff
@@ -22,4 +22,14 @@ module System
   def self.cpu_count
     Crystal::System.cpu_count
   end
+
+  # Returns the short user name of the currently logged in user or `nil` if it
+  # can't be determined. Note that this information is not secure.
```

Comment:
This is what happens when `getlogin` returns a `NULL`: https://carc.in/#/r/2hbo. You need to manually convert it to a `nil`, `String.new` won't do that for you.